src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/DirectionUploadOrderAboutToSubmitHandler.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.handler;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.finrem.caseorchestration.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.controllers.GenericAboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.finrem.caseorchestration.error.CourtDetailsParseException;
+import uk.gov.hmcts.reform.finrem.caseorchestration.mapper.FinremCaseDetailsMapper;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.EventType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CaseType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.FinremCaseData;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.FinremCaseDetails;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.AdditionalHearingDocumentService;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Slf4j
+@Service
+public class DirectionUploadOrderAboutToSubmitHandler extends FinremCallbackHandler {
+
+    private final AdditionalHearingDocumentService service;
+
+    public DirectionUploadOrderAboutToSubmitHandler(FinremCaseDetailsMapper finremCaseDetailsMapper,
+                                                    AdditionalHearingDocumentService service) {
+        super(finremCaseDetailsMapper);
+        this.service = service;
+    }
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, CaseType caseType, EventType eventType) {
+        return CallbackType.ABOUT_TO_SUBMIT.equals(callbackType)
+            && CaseType.CONTESTED.equals(caseType)
+            && (EventType.DIRECTION_UPLOAD_ORDER.equals(eventType));
+    }
+
+    @Override
+    public GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle(FinremCallbackRequest callbackRequest,
+                                                                              String userAuthorisation) {
+        FinremCaseDetails caseDetails = callbackRequest.getCaseDetails();
+        String caseId = String.valueOf(caseDetails.getId());
+        log.info("Invoking contested event {} mid callback for case id: {}",
+            EventType.DIRECTION_UPLOAD_ORDER, caseId);
+        FinremCaseData caseData = caseDetails.getData();
+
+        List<String> errors = new ArrayList<>();
+        log.info("Storing Additional Hearing Document for Case ID: {}", caseId);
+        try {
+            service.createAndStoreAdditionalHearingDocuments(caseDetails, userAuthorisation);
```

Comment:
We have duplicate createAndStoreAdditionalHearingDocuments which is defined in HearingDocumentController.java and endpoint is /contested-upload-direction-order

Reply:
I am moving into handler once this tested - and merged - will remove the old endpoint.